### ISSUE TITLE
discussion: monorepo structure for v7

### ISSUE THREAD

**Issue by DanielRuf** [foundation/foundation-sites#11847]
## Description

Foundation v7 will likely be a monorepo and contain different packages. Compiler, design system, theme & custimizer, the different components, plugin system.

We might also create codemods for easier migration of a few things plus some foundation-migrate.js which outputs console information for migrating current code.

## Possible Solution

...

## Checklist

- [x] I have read and follow the CONTRIBUTING.md document.
- [x] There are no other issues similar to this one.
- [ ] The issue title and template are correctly filled.

**Comment by SassNinja** [comment]
@DanielRuf do you already have a solution in mind for managing the packages? maybe [lerna](https://github.com/lerna/lerna)?

**Comment by DanielRuf** [comment]
Either lerna or yarn.

**Comment by DanielRuf** [comment]
I propose lerna with yarn.

**Comment by DanielRuf** [comment]
I think we can already experiment with the new monorepo and move some current parts to it (plugins for example).

**Comment by paulfelton** [comment]
Has anyone considered whether it might be possible to generate web components as a target for foundation's components?  Or stencil components that could be transpiled?

This could make it easier for components to be versioned individually, and incorporated in the end-designer's design systems (regardless of the technology they base it on).  Likewise for the developer, they could use the components regardless of whether they're generating an Vue, Angular, React (to a certain extent) or vanilla site?

**Comment by DanielRuf** [comment]
> Has anyone considered whether it might be possible to generate web components as a target for foundation's components? Or stencil components that could be transpiled?

This is one of our plans.

**Comment by DanielRuf** [comment]
I added the link to semantic-release.
In the future we should publish only on npmjs (and additionally on GPR) - GitHub Package Registry).

Other frameworks and platforms can and should consume npm packages or unpkg files.

**Comment by paulfelton** [comment]
Out of curiosity, do you have a roadmap to version 7, or aspirations?  I've seen the project board.

**Comment by DanielRuf** [comment]
@paulfelton we had a small one but I would say we should revise / update it.

https://github.com/foundation/foundation-sites/wiki/Project-Roadmap

**Comment by DanielRuf** [comment]
@joeworkman to which platforms do we currently / until now publish to? I want to check if and how they can consume npm packages and if we can publish only to npmjs in the future.

Bower for example registered the old foundation-sites package to the zurb organization.

**Comment by joeworkman** [comment]
We will definitely start to organize our thoughts around F7 very soon. It will be a lot of work and hopefully we can get more awesome devs to contribute. 

@DanielRuf Right now I only publish to NPM and RubyGems. We used to publish to meteor but that was stopped a long time ago. For F7, we can probably pair that down to just NPM. 

As for the nom organization. I sent their support team multiple requests and never heard back. For F7, it would definitely be nice to have our own org. I will keep pestering them. Now that they are ran by Github, it may be easier.

**Comment by joeworkman** [comment]
This issue has been mentioned on **Foundation Open Source Community**. There might be relevant details there:

https://foundation.discourse.group/t/foundation-and-css-variables/2469/4

**Comment by Geddo** [comment]
css-variables would be awesome. I integrated it for a bunch of objects like this:

_settings.scss:
`:root{ 
  --current-color: var(--primary-color, var(--default-color,#FF0000));
}`

So if the css-variable --primary-color is not set, I use the default color #FF0000 for it. For all other elements which should use the new css-variable I created a mixin then like this:
`.header {
    background: $primary-color;   
    @include colorfrompage("background-color");
}`

or like
`.h1{
    @include colorfrompage("color");
}`

And this is my simple mixin:
`@mixin colorfrompage($property) {
    transition: all 0.25s;
    #{$property}:  var(--current-color);
}`

So with that I can easly set a new primary-color via a css-variable in my pages head and with additional scripts like swup.js I get smooth page and color transitions :-)

**Comment by joeworkman** [comment]
CSS custom properties are definitely Italy something g that we want in v7. I have already started a modified version of v6 that supports them already. Although, it’s just a proof of concept. A big change like this should happen in v7.

**Comment by ettoredn** [comment]
If I may suggest:
1. focus on the grid system / SCSS framework and release an early version with that only
2. drop as many plugins as there are already well working alternatives for them
3. build the remainder plugins as *reusable* web components using StencilJS and TypeScript

**Comment by dlewand691** [comment]
For consideration, using CSS variables, this technique looks pretty intriguing as a way to write less code. 
https://propjockey.github.io/css-media-vars/

**Comment by joeworkman** [comment]
I actually already have a build that uses CSS variables. Its not perfect but it's definitely the way to go moving forward. The side effects is that we cannot use Sass color math for many things when using CSS variables. But a lot of effects can be achieved with CSS filters and other modern css techniques.

**Comment by paulfelton** [comment]
I'd vote for [stenciljs](stenciljs.com) too, possibly then documenting with [Storybook](https://storybook.js.org/)

**Comment by HansUXdev** [comment]
It's been 3 years now, bootstrap is in alpha and the core maintainers seem overwhelmed (understandably).
On the positive side, it seems like most of the issues with zurb going hands off are largely resolved finally but I'm sure there are still broken links here and there?

I don't know about anyone else, but I feel like there needs to be a round table / steering committee discussion of sorts scheduled to jump start V7. So here is my suggestion. 

1. We schedule a date sometime around the end of august or september for an open zoom meeting that meets on a regular basis with an agenda. 
2. We publish steering meetings to youtube just like Node does.
3. We use social media and the Forums to reengage our community by running polls on various features for V7 
(ex: scale 1-4 how to you feel about ____ feature?, followup, would you contribute to this?). 
4. We start a private monorepo until alpha that is invite only to people on the steering committee.
5. We solidify our build process, CI, documentation, project board(s), etc to make this more inviting to everyone.
6. We start with a SCSS library building off the awesome utilities that foundation already has and maybe taking inspiration from other frameworks and libraries like tailwind.
7. We then move on to scss components (building blocks).
8. We start building the JavaScript components.
9. We start building the tooling around the new framework (boilerplates, layouts/templates, CLI, VS-Code Extensions, etc)

**Comment by DanielRuf** [comment]
> 1. We schedule a date sometime around the end of august or september for an open zoom meeting that meets on a regular basis with an agenda.

Who is "we"? Currently only Joe and I are actively working on Foundation.

> 2. We publish steering meetings to youtube just like Node does.

Much effort and work for such a small project in my opinion.
In general we already have a roadmap for a long time. But no time besides work and other projects.

> 4. We start a private monorepo until alpha that is invite only to people on the steering committee.

Not a good idea, development will be always in the public. No private repos and no invite only stuff.


> 5. We solidify our build process, CI

Both work, I see no reason to change anything.

Not sure but most of these points are either not relevant, need more active contributors or repeat what we already have in the roadmap and in the relevant issues.

**Comment by HansUXdev** [comment]
> Who is "we"? Currently only Joe and I are actively working on Foundation.

@DanielRuf I don't know if you are intentionally condescending and rude or if it's an accident. Honestly I'm done caring.
I've seen a lot of this and not just with comments to me but to few others as well, especially to people just trying to learn.
"We" obviously refers to at-least the 8 others in this conversation. Making comments like this doesn't invite collaboration, it just exclusionary.

There are more than a few people trying to get more involved in foundation other than just you and Joe.

> Much effort and work for such a small project in my opinion.

We're all grateful for what you and Joe do but Foundation isn't a small project. 
Setting up a meeting, running it and putting it on youtube isn't a lot of work and I'm more than happy to do it. In fact, in my opinion it's far more fruitless to talk about this on github issues for 3 years... Also foundation is  2 frameworks and multiple libraries, boilerplates, etc. That's not a small project, its much bigger than bootstrap actually.

Also it just took over a month to publish a simple fix to the CLI which is honestly the most powerful part of the entire ecosystem. I maintain a few npm libraries both here on my current Github and with my work account.

The whole point of what I'm suggesting is to be constructive and help get more active contributors and re-engage the community so that rather than building out features on a roadmap others suggested and abandon, we take a fresh look and start with DOING it. **Nothing about your comment is constructive to your own end goals.**

I've volunteered to help on multiple points with what I'm interested in fixing and have the time to contribute to.
If you guys want to put all the weight on yourself and complain about it or shut people out, I'm not going to stop you. 
I'll just stop trying because that's a great way to kill an open source project and I've seen this happen before...

Honestly, why would I even want to submit a PR when someone makes these kind of condescending comments?
This is why I've maintained my own internal branches of foundation for sites, email and the cli for the last 3 years rather than submitting a PR.

**Comment by DanielRuf** [comment]
> Also it just took over a month to publish a simple fix to the CLI

Please keep in mind that Joe and I do this in our free time and for free. Just saying.

**Comment by DanielRuf** [comment]
> The whole point of what I'm suggesting is to be constructive and help get more active contributors and re-engage the community so that rather than building out features on a roadmap others suggested and abandon, we take a fresh look and start with DOING it. **Nothing about your comment is constructive to your own end goals.**

To be honest, what would help us at the moment would be to have more reviews on open PRs, PRs for open issues and issues resolved so we can make a clear cut and start the work on v7. Nothing against you or others, there is just still much to do before we can move on. And to decide. And most things are already decided on the original roadmap + some decisions and ideas mentioned here. Ideas are very welcome but I know that most of the volunteers can and will not attend such meetings - we tried that in the pst with all Yetinauts and it didn't really work.

If you want to organize this (date, time, people, accounts, ...) you are very welcome to do so and help usw. I do not have the time for this and I speak from my past experience (Nodejs org, Foundation and others) and it's not that easy.

**Comment by DanielRuf** [comment]
> Honestly, why would I even want to submit a PR

PRs are always reviewed when I find some time. And we merged some in the last weeks. We have to get the other projects forward too (Foundation emails got some new releases, you might have noticed that).

**Comment by DanielRuf** [comment]
> This is why I've maintained my own internal branches of foundation for sites, email and the cli for the last 3 years rather than submitting a PR.

Gatekeepig will not help us at all to bring this forward togehter. I propose you submit some PRs, we check if the proposed changes would be good and we can plan them for the next releases.

**Comment by HansUXdev** [comment]
I'm not Gatekeeping, it was just the opposite. what I did was in response to gatekeeping and also for organizational needs (full-stack boilerplates) or educational purposes for students because it's not uncommon for companies to maintain their own branches.

As for PR's I'd be happy to review existing PR's test them and either close them or recommend a merge. But if you are looking for new PR as they relate to https://github.com/foundation/foundation-sites/milestone/40, that's not something I can commit to, Trust me, if I could commit to them, I would. 

Almost all of them jQuery related issues and I'm just not as familiar with the architecture the JS side of this framework and quite honestly, I have zero interest in it because I don't use it. I use foundation SCSS as a utility lib for react and other things. I'm also spending a good 4-6 hours writing open source curriculum every day with baby who's just walking around.  

We all have other priorities at work and at home but I don't believe that PR's or the lack there of should roadblock implementing V7 after a 3 year delay...

**Comment by HansUXdev** [comment]
>   I know that most of the volunteers can and will not attend such meetings - we tried that in the pst with all Yetinauts and it didn't really work.
If you want to organize this (date, time, people, accounts, ...) you are very welcome to do so and help usw. I do not have the time for this and I speak from my past experience (Nodejs org, Foundation and others) and it's not that easy.

I'm guessing you weren't with foundation around 2015-2016? 
We had I think 10+ volunteers on the conference calls for foundation 6 alpha, etc.
But yeah, I'm happy to help with that because I have more experience doing it and the time available.
The first thing to do would be to talk to the people managing the social media, then reach out to Yetinauts and get their availability and interests.

**Comment by DanielRuf** [comment]
> As for PR's I'd be happy to review existing PR's test them and either close them or recommend a merge. But if you are looking for new PR as they relate to [foundation/foundation-sites/milestone/40](https://github.com/foundation/foundation-sites/milestone/40), that's not something I can commit to, Trust me, if I could commit to them, I would.

We welcome reviews and contributions for current and new PRs too. You do not have to contribute new PRs if the issues are too complicated / not that easy to solve.

> We all have other priorities at work and at home but I don't believe that PR's or the lack there of should roadblock implementing V7 after a 3 year delay...

Well, users of the other projects ask also for updates so we fight on multiple fronts. Like I wrote, Foundation Emails got a few new releases and other projects too. We should resolve the big issues there before we move on - especially as it makes not much sense to open a big new project (v7) and directly drop support for v6 (with such few active people helping with the current tasks).

And I'm also active in other projects ao I can not invest so much of my free time.

> We had I think 10+ volunteers on the conference calls for foundation 6 alpha, etc.

Currently all other Yetinauts are not actively involved anymore since the move to a community project. Maybe you might have a better connection to them.

When I joined Nicolas tried to do these meetings but after a while we stopped doing them because no one joined or had time.

**Comment by DanielRuf** [comment]
> As for PR's I'd be happy to review existing PR's test them and either close them or recommend a merge. But if you are looking for new PR as they relate to [foundation/foundation-sites/milestone/40](https://github.com/foundation/foundation-sites/milestone/40), that's not something I can commit to, Trust me, if I could commit to them, I would.

We welcome reviews and contributions for current and new PRs too. You do not have to contribute new PRs if the issues are too complicated / not that easy to solve.

> We all have other priorities at work and at home but I don't believe that PR's or the lack there of should roadblock implementing V7 after a 3 year delay...

Well, users of the other projects ask also for updates so we fight on multiple fronts. Like I wrote, Foundation Emails got a few new releases and other projects too. We should resolve the big issues there before we move on - especially as it makes not much sense to open a big new project (v7) and directly drop support for v6 (with such few active people helping with the current tasks).

And I'm also active in other projects so I can not invest so much of my free time.

> We had I think 10+ volunteers on the conference calls for foundation 6 alpha, etc.

Currently all other Yetinauts are not actively involved anymore since the move to a community project. Maybe you might have a better connection to them.

When I joined Nicolas tried to do these meetings but after a while we stopped doing them because no one joined or had time.

**Comment by ettoredn** [comment]
I have been wondering: why nobody is interested anymore in Foundation? Are better alternatives out there? Does anyone need a CSS grid framework anymore?

Don't tell me Bootstrap is an alternative. Bootstrap is trash.

**Comment by HansUXdev** [comment]
> I have been wondering: why nobody is interested anymore in Foundation? Are better alternatives out there? Does anyone need a CSS grid framework anymore?
> 
> Don't tell me Bootstrap is an alternative. Bootstrap is trash.

In my humble opinion,
Foundation has always been less popular compared to bootstrap in part because originally, it was just a css style guide, or the "foundation" for responsive sites. Bootstrap actually came later and used foundation.

Foundation started dying off after Zurb had a massive layoff and most of the contributors left. 
But just like F4A was bad timing with using angular 1 and angular 2 releasing shortly after, F6 was also poor timing because jQuery was dying and css flex and grid were coming out. As a result F6 has 3 different grid systems, some of which also came (in part) from F4A which has since completely died. Also around the same time foundation for email 2 was release.

Once css grid started trending a lot of people were talking about how frameworks were going to die because complex layouts are so much easier to create now. This was commented a lot during some of the early roadmap discussions for V7. In this light, all major frameworks have seen increasingly less popularity and the demand for dropping jQuery has been a big part of this.

In other words it's in an awkward transition between web pages being the standard and the new web standard of web components.

Bootstrap isn't bad, it just doesn't have the same amazing tooling or ecosystem behind it like Foundation does.
Recently I just started building a [boilerplate for Bootstrap V5](https://github.com/HansUXdev/Bootstrap5-Rapid-Design-Boilerplate) that has virtually no configuration, automatic cross-browser testing, and some other goodies I haven't gotten around too.

**Comment by ettoredn** [comment]
Am I wrong in stating that Bootstrap is more oriented towards prototyping while Foundation is about building actual sites?

What are the exact use cases Foundation is supposed to help with?

**Comment by ettoredn** [comment]
> In other words it's in an awkward transition between web pages being the standard and the new web standard of web components.

I wouldn't go as far as calling it a transition. WC fit enterprise design systems well, but I'm not so sure about small and mid size projects. I'm trying to use them through StencilJS right now on a project, and unless you just go full framework like Angular/React (by this I mean you make everything a component because why not), design choices (what should this component do? what should be componentized and what not?) are not so clear, so far.

Case in point: try to use shadow DOM and use Foundation's grid (and the grid only via @include) in multiple component.. Look at that, so many duplicated CSS. Very nice for page performance.

**Comment by HansUXdev** [comment]
Actually, I would say it's the reverse. Zurb created foundation as an in house tool to go from prototype to production because they are a design studio and focuses heavily on design feedback and progressive design. The reason they went hands off is to focus on Helio.app, a design feedback product originally broken down into 4 apps (some built off F4A if I'm not mistaken)...

You can use foundation for either prototypes of complete websites. I've built probably about 50 sites with it going back to F5 and about 20 email templates the email framework.

When I was teaching coding one of things I did was print out some of the core components and a dozen building blocks. Then I had them use them as paper prototypes to collect feedback on the designs with other teams for their projects. Once the designs were validated they then used the CLI to download the blocks and build the final site in under 2 hours because they had planned everything out.

Those were high school kids btw...

I haven't used StencilJS but I've had my eye on it and I used polymer in the past. Shadow DOM has always been awful and thats part of why react is so popular. It solve the problem of scoped styles way more simply.

**Comment by pboling** [comment]
I have a very narrow comment on the mono repo idea.  Want to make sure that having separate repos for plugins is given full consideration.  Generally the benefit would be forcing a first-class plugin system, which both official and third party plugins must use.  Keeping the plugins inside the mono repo might discourage third party development, or worse, prejudice the tooling in ways that make it more difficult to use a third party plugin than a built-in.

This comment may be quite irrelevant though, given how little I know of the plugin architecture, and having the official plugins developed as part of a holistic core release is also appealing.

Thanks for building Foundation!  If my dev path had turned down the front end road I'd be jumping in to help more.

**Comment by Baggio89** [comment]
Count me in... im with Foundation for 6+ years now. Can help with different things... experience: js, jquery, es(x), css, sass, html, float-g, flex-g, css-g, npm, yarn, node, webpack, wordpress, php, and many more. There must be something i can do 💯 . 

Let make this Christmas a good one :)

**Comment by Baggio89** [comment]
Seems really dead now. Even bootstrap is catching up with Bootstrap 5.

**Comment by DanielRuf** [comment]
@Baggio89 feel free to review some PRs and issues. But at the moment I am not that much involved in Foundation Sites anymore.

**Comment by Baggio89** [comment]
That's a pity. Shouldn't someone else take over then? It's such a waste not releasing a new foundation sites.

**Comment by joeworkman** [comment]
One of my goals for 2021 is to make good headway on F7. I would like to build a small core group that would like to help push it forward. I am finally wrapping up a couple of big projects from 2020. I have reserved time in Q2 to start working on this more. If you would like to be involved in the process, let me know.

**Comment by dlewand691** [comment]
Hi @joeworkman. I'm a huge fan of Foundation and been using it since about version 3. I work for Cognizant and it has been our framework of choice for the last few years. Our company has over 280,000 employees in 37 countries so we rely heavily on the flexibility and speed of rapid prototyping that Foundation gives us. I'd like to volunteer to help out wherever I can, mainly on the CSS development. 

I've created a framework for our dev team that expands on Foundation's capabilities and would love to get your (and others) thoughts on it. Take a look at this repo: 

https://github.com/dlewand691/_COG-Framework-2020

A high-level wish list of things for V7 I'd like to request/suggest/discuss are:

- Use CSS custom properties instead of Sass variables
- Change `@import` to `@use` or `@forward`
- Use CSS Grid (there's already a branch of this from 2017 but needs work)
- Use native HTML elements in place of custom ones where possible. `<details>` could replace accordions, dropdowns and modals?
- Should plugins like Orbit be removed in favor of more advanced ones like Swiper?
- Integrate PostCSS to purge unused CSS and other transformations
- Look at other frameworks like Bulma and Codyhouse to get any inspiration for how they're doing things

Anyhow, please let me know your plans and how I can help. I don't want to see Foundation die, it's too valuable a resource! Look forward to hearing your thoughts. Thanks!

**Comment by rahiyansafz** [comment]
what's new @joeworkman ??? when will V7 come?

**Comment by kylek14** [comment]
> One of my goals for 2021 is to make good headway on F7. I would like to build a small core group that would like to help push it forward. I am finally wrapping up a couple of big projects from 2020. I have reserved time in Q2 to start working on this more. If you would like to be involved in the process, let me know.

Out of curiosity, what help are you looking for? Interested in helping but more on the supporting side (webdev, docs, etc.)

**Comment by IamManchanda** [comment]
Very Good Morning, In this Modern Web Tech, If Foundation 7 has to become a reality

This is what we need to do

1. Make Foundation 7 CSS/SCSS on par with Bootstrap 5 CSS/SCSS
2. Remove jQuery Dependency, docs should contain not only html examples but also `Vue`, `React` & `Angular` examples.
3. Just-in-Time Mode https://tailwindcss.com/docs/just-in-time-mode 
4. Make [Foundation Prototyping Utilities](https://get.foundation/sites/docs/prototyping-utilities.html) to be on par with Tailwind CSS extensive list of classes and make them available by default with `Just-in-Time Mode`

**Comment by qeet** [comment]
@IamManchanda 

I just released a library that lets you use all the Tailwind utilities with the current version of Foundation. All you need to do is include a tiny (~8Kb) Javascript in your page and then you can use all Tailwind utilities to style your Foundation components.

See: https://github.com/qeet/foxtailcss for docs & code

**Comment by DaSchTour** [comment]
@IamManchanda instead of reinventing the wheel maybe foundation should build on top of tailwind and simply provide some nice UI components. There is a lot of effort in building something like tailwind and simply rebuilding it would not do the trick. It will always fall behind tailwind.

**Comment by DanielRuf** [comment]
> @IamManchanda instead of reinventing the wheel maybe foundation should build on top of tailwind and simply provide some nice UI components. There is a lot of effort in building something like tailwind and simply rebuilding it would not do the trick. It will always fall behind tailwind.

Exactly. We know how reinventing the wheel went in the past (orbit, grids, ...) which ended in more problems and work than a small team could do.

**Comment by Baggio89** [comment]
Tailwind, Alpine JS and Laravel Mix, ready, set, go!

**Comment by dlewand691** [comment]
Madosel is a great new framework, currently in alpha, but could be a great inspiration for next steps on Foundation. Take a look here: [https://www.madosel.com/](https://www.madosel.com/)

**Comment by pboling** [comment]
[Svelte](http://svelte.dev) is the next major framework in Javascript, so integration with Svelte (which already has first class Tailwind support) should be a goal of a future Foundation. Svelte will eat much of React, Vue and Angular over the next couple years.

**Comment by IamManchanda** [comment]
> @IamManchanda instead of reinventing the wheel maybe foundation should build on top of tailwind and simply provide some nice UI components.

I am cool too ... yes reinventing the wheel wouldn't make sense with short team

**Comment by joeworkman** [comment]
I know that Tailwind is the cool new(-ish) kid on the block. However, I do not see Foundation 7 will be going that way. Utility classes are useful. But there are limits to how far I feel they should go. F7 will be blazing new trails.

**Comment by joeworkman** [comment]
I will be doing another hangout soon if anyone wants to be a part of that... https://foundation.discourse.group/t/foundation-7-hangout-looking-forward-to-f7/

**Comment by rahiyansafz** [comment]
> I will be doing another hangout soon if anyone wants to be a part of that... https://foundation.discourse.group/t/foundation-7-hangout-looking-forward-to-f7/

Please Joe! make foundation Great again.

**Comment by joeworkman** [comment]
I have another hangout setup for next week for those that can make it... https://foundation.discourse.group/t/july-hangout-for-foundation-7/4611

**Comment by edoardo-bluframe** [comment]
Hey everybody!

**Mega** upvote for:

- CSS Grid
- ES6 Only / Bye Bye jQuery

Can I help as a dev?

**Comment by maietta** [comment]
Hi @adventdesignllc, I'd love to see a version 7 released as well. The project is certainly not dead. It's just a really stable framework.

I have been using Foundation 6.5+ with Vue 3 + Vite for some time now and even build a project base around it at https://github.com/PremoWeb/SDK-Foundation-Vue that is worth checking out if you want to adopt Foundation with Vue 3.

I would absolutely love to devote all my time to open source projects instead of working on client projects but client projects are what pay the bills. I'm seriously considering spending some time before fall and winter to drum up enough sponsorship so I can make the transition to working on projects like Foundation full time. The sooner we have people who can devote serious hours to the project, the sooner everyone benefits. I still feel that Foundation for Sites is the very best front-end framework for modern websites though I do feel there is room for improvement, especially from a web developer's perspective.

It'd be rad to drum up about $4-5k/mo which is nothing these days to devote my time to this. I am by no means a front-end developer but I do have nearly 10 years of experience using Foundation. (Wow, has it really been that long?) I know that I was using Foundation for some time before attending the Foundation 5 software launch party back in 2013. That was an amazing experience getting to meet the people behind the project and pick their brains for all that juicy info about web design, goals, the vision of the future, etc. Those people really know their stuff.

One thing I can offer to the table is complete tooling for every platform (Linux, Windows & Mac), and the ability to integrate into existing frameworks like Vue 3, React, Angular, etc. I can build a website from top to bottom that can replace the existing, overly complicated get.foundation system which I think is built upon proprietary hosting (or at least, it used to be).

My current Vue + Foundation template mentioned above does give me that live hot-reloading capability that first got me interested in the SASS version using Gulp, etc. I had nothing but problems trying to use Foundation's provided tools as things were evolving way too fast. Bowser, Yarn, Gulp, these were all new to me at the time so that was probably my issue.

I've got a version of Foundation that works with Webpack somewhere, but I ran into a strange edge case issue that I couldn't easily resolve, which caused me to look at Vite build system instead, which did solve my issue. 

Foundation 7 should certainly be built to operate easily within existing build systems by default. It should be easier to configure and there should be a fresh website with an improved catalog of ready-to-use components.

Mixins should be easier to understand and use. To this day, I still can't figure out how to get mixins working properly within my Vite system. With more time spent on this, I could probably have this hammered out.

Anyway, I don't know what's up with the status of 7, as it's been MANY YEARS since it was first discussed. I really hope we can get some serious interest in it. So far, I really only see one person who works a lot on this project but my guess is he's got to pay his bills too and I don't think this project pays anything at this point. The joys of Open Source!

**Comment by adventdesignllc** [comment]
I'd certainly be willing to contribute monthly support to help development. I myself would be happy to chip in some hours, though I'm not sure the depth of my skills would be deep enough to be helpful.

@maietta I'm not familiar with Vue but looks like I should look into. Is some sort of instantaneous live-update of sass changes?

Foundation is working great for me at the moment. I understand HTML and SCSS, but like you, Yarn, Gulp etc. are the bane of Foundation based work. I don't understand any of that command line functionality, and have 'broken' the Foundation build system many times over the years, resulting in hours and hours of beating my head against the wall trying cut and paste commands to get the system back up to compile the projects. I've got a core project I duplicate and run "npm start" and it works. "Foundation New" command hasn't worked for me for ages.

I wonder how many front-end coders like me would be more apt to use a system like Foundation if the way to get a project set up and compiled wasn't so archaic.

**Comment by maietta** [comment]
@adventdesignllc, Firstly, thank you. A willingness to contribute both financially, even a little bit as well as some hours, even if you don't think you're qualified, are both extremely important in my opinion. That is a seed worth planting.

I fully agree that the Foundation SASS edition must be easy to get set up and use, regardless of the operating system or skill level. It was absolute hell for me to get started with it and I'm no spring chicken in web development. Yet I kept going back to the precompiled ZIP or CDN options due to the many problems I kept running into. Thankfully, I broke past this issue and I'd love to see other people adopt my template and improve upon it.

In desperate of needing more action and less talk, I've gone ahead and created a new Github Organization called "Foundation Remix" which will rebuild the Foundation 6.5.x ecosystem using modern cross-platform but minimal tooling, a complete rebuild of the website in Vue 3, tooling, distribution channels, etc. I'll stand up a dedicated community forum devoted to moving to 7 and I'll start assembling talking points from the existing Foundation discussions and issues as it relates to 7, including suggestions for direction, etc.

If you are serious about this as I am, shoot me an email at nick@premoweb.com. I'm going to dig around here to see who else may be interested in following along or to help make this happen quickly. Again, it's been YEARS since 7 was first discussed. Time for action.

**Comment by joeworkman** [comment]
@adventdesignllc I have started a discussion thread about F7 and F8 roadmaps. I am not sure if you have seen it... https://github.com/foundation/foundation-sites/discussions/12310

This year has threw me some unexpected challenges, however, I do still feel that a beta for F7 can happen in 2023 as I estimated in my roadmap. Until then I will keep shipping patch updates to F6. I released [v6.7.5](https://github.com/foundation/foundation-sites/discussions/12449) last month. 

There is a lot more to Foundation than just this repo. There is a lot of underlying repos that make a lot of the magic happen. I have also been spending some time getting some of these updated or completely replacing them with something new. For example, this year I released [Proton](https://github.com/foundation/proton) as a future replacement for panini. I feel that it's a much more powerful, elegant and easier to maintain solution. 

@maietta I look forward to seeing what you do with Foundation Remix 🎉

**Comment by maietta** [comment]
Hi @joeworkman, thanks for the encouragement and I totally understand about unexpected life challenges. For me, they are less frequent than they used to be so I'd like to leverage that. I would much rather put my energy into the existing Foundation but feel that many of the things I want to do will fall outside what this project is aiming for. Perhaps there could be a happy medium here. I am still investigating how I can be of use.

I've read over the roadmap discussion for 7 and 8 and agree with most of it. Perhaps I can be working on knocking off some of the items on the checklist. Admittedly, I'm going to need to get better at Typescript, web components, et cetera.

Proton looks interesting, although I'm not a user of the Pug template system, I can see how it's useful.

As of right now, Foundation Remix is just an idea and will be very much a collection of experiments until such time as it can prove itself viable. If anything, I want it to be helpful to Foundation, not necessarily a replacement. I come from more than two decades in backend development, not front-end, so this is a bit overwhelming, actually.

I'm thinking of porting the Javascript components to Typescript, going native to remove any need for jQuery. If I can do this without bugs, that'd be amazing and I think a good way to help this project along.

**Comment by maietta** [comment]
I gave up on the project.

I moved away from legacy PHP projects and moved to adopt Tailwind. I'm sorry.

**Comment by pboling** [comment]
It's ok to let it go.  Thanks for the efforts 🙇

**Comment by matteorchard** [comment]
Ah the memories! Thanks F6 for kickstarting my career!

**Comment by TheDigitalOrchard** [comment]
After years of trying different layout frameworks, I'm considering now moving to Foundation as a clean and sane "foundation". Bootstrap has been my main go-to for the past several years, but I want something that's less ... what's the word ... opinionated?

In an ideal world, I could combine multiple layout frameworks together on a single page, where some components use one and other components use another. I'm sure that's entirely possible with limited success, but it also feels messy. But choosing just one also feels like a sacrifice in some ways. That's my current struggle.  _(I know Web Components directly address this.)_

Do you envision Foundation 7 being be a smooth upgrade from Foundation 6, or a jarring experience? Since we're at v6.9 now, I'm imaging F7 to be coming ... soonish?

**Comment by clizlee** [comment]
@joeworkman is there any update on F7 progress? Or is this being discussed on discord now?

**Comment by joeworkman** [comment]
I do not have an ETA at this time. I have done work on it. For example, I do have a version that uses CSS custom properties. 

I've been developing a new CMS, called Total CMS. It's taken a lot longer than I expected but it's now in beta. I'll be able to shift focus back to Foundation after that ships and is stable. 

I have a commercial product that relies on Foundation. You can see it at https://www.foundationstacks.com/. I want to release a major version of that next year, which means that I'll need to make way on updating the framework. 

Hope that helps. Happy to discuss more here or on Discord (probably better, if you are there)
